neuralmonkey/trainers/delayed_update_trainer.py

```diff
@@ -0,0 +1,244 @@
+from typing import Dict, List, Tuple
+# pylint: disable=unused-import
+from typing import Optional
+# pylint: enable=unused-import
+
+import tensorflow as tf
+from typeguard import check_argument_types
+
+from neuralmonkey.decorators import tensor
+from neuralmonkey.runners.base_runner import (
+    Executable, ExecutionResult, NextExecute)
+from neuralmonkey.trainers.generic_trainer import (
+    GenericTrainer, Objective, Gradients)
+
+
+class DelayedUpdateTrainer(GenericTrainer):
```

Comment:
Tohle pořebuje docstring, co vysvětlí, co to je a jak jsou tam ty fáze.

Reply:
done